adafruit_max31856.py

```diff
@@ -0,0 +1,306 @@
+_MAX31856_FAULT_TCHIGH = const(0x08)
+_MAX31856_FAULT_TCLOW = const(0x04)
+_MAX31856_FAULT_OVUV = const(0x02)
+_MAX31856_FAULT_OPEN = const(0x01)
+
+
+class ThermocoupleType: # pylint: disable=too-few-public-methods
+    """An enum-like class representing the different types of thermocouples that the MAX31856 can
+    use. The values can be referenced like ``ThermocoupleType.K`` or ``ThermocoupleType.S``
+    Possible values are
+
+    - ``ThermocoupleType.B``
+    - ``ThermocoupleType.E``
+    - ``ThermocoupleType.J``
+    - ``ThermocoupleType.K``
+    - ``ThermocoupleType.N``
+    - ``ThermocoupleType.R``
+    - ``ThermocoupleType.S``
+    - ``ThermocoupleType.T``
+
+    """
+    # pylint: disable=invalid-name
+    B = 0b0000
+    E = 0b0001
+    J = 0b0010
+    K = 0b0011
+    N = 0b0100
+    R = 0b0101
+    S = 0b0110
+    T = 0b0111
+    G8 = 0b1000
+    G32 = 0b1100
+
+
+class MAX31856:
+    """Driver for the MAX31856 Universal Thermocouple Amplifier
+
+      :param ~busio.SPI spi_bus: The SPI bus the MAX31856 is connected to.
+      :param ~microcontroller.Pin cs: The pin used for the CS signal.
+      :param ~adafruit_max31856.ThermocoupleType thermocouple_type: The type of thermocouple.\
+      Default is Type K.
+
+    """
+
+    # A class level buffer to reduce allocations for reading and writing.
+    # Tony says this isn't re-entrant or thread safe!
+    _BUFFER = bytearray(3)
+
+    def __init__(self, spi, cs, thermocouple_type=ThermocoupleType.K):
+        self._device = SPIDevice(spi, cs, baudrate=500000, polarity=0, phase=1)
+
+        # assert on any fault
+        self._write_u8(_MAX31856_MASK_REG, 0x0)
+        # enable circuit faults
+        self._write_u8(_MAX31856_CR0_REG, _MAX31856_CR0_OCFAULT0)
+
+        # set thermocouple type
+        # get current value of CR1 Reg
+        conf_reg_1 = self._read_u8(_MAX31856_CR1_REG)
+        conf_reg_1 &= 0xF0  # mask off bottom 4 bits
+        # add the new value for the TC type
+        conf_reg_1 |= int(thermocouple_type) & 0x0F
+        self._write_u8(_MAX31856_CR1_REG, conf_reg_1)
+
+    @property
+    def temperature(self):
+        """The temperature of the sensor and return its value in degrees celsius. (read-only)"""
+        self._perform_one_shot_measurement()
+
+        # read three bytes starting at the MSB of the
+        raw_read = self._read_u24(_MAX31856_LTCBH_REG)
+
+        # fix sign if needed
+        if raw_read & 0x800000:
+            raw_read |= 0xFF000000
+
+        # bottom 5 bits are unused
+        raw_read >>= 5
+        # scale back from linearized value to temp
+        temp_float = raw_read * 0.0078125
+
+        return temp_float
+
+    @property
+    def reference_temperature(self):
+        """The temperature of the cold junction in degrees celsius. (read-only)"""
+        self._perform_one_shot_measurement()
+        raw_read = self._read_u16(_MAX31856_CJTH_REG)
+        cold_junction_temp = raw_read / 256.0
+
+        return cold_junction_temp
```

Comment:
This will likely have a similar issue for negative temperatures as the `temperature` func above.

Reply:
done

Reply:
done